include/picongpu/plugins/binning/axis/LogAxis.hpp

```diff
@@ -0,0 +1,287 @@
+                std::vector<T> binWidths;
+
+                /**
+                 * @TODO store edges? Copmute once at the beginning and store for later to print at every
+                 * iteration, also to be used in search based binning
+                 */
+                struct LogAxisKernel
+                {
+                    /** Function to place particle on axis, returns same type as min and max */
+                    T_AttrFunctor getAttributeValue;
+                    /**
+                     * logMin and logMax values in the range of the binning. Values outside this range are
+                     * placed in overflow bins
+                     */
+                    T logMin, logMax;
+                    /** Number of bins in range */
+                    uint32_t nBins;
+                    /** Using type depending on whether T is integer or floating point type to avoid precision loss */
+                    using ScalingType = std::
+                        conditional_t<std::is_integral_v<T>, std::conditional_t<sizeof(T) == 4, float_X, double>, T>;
+                    ScalingType scaling;
+                    /** Enable or disable allocation of extra bins for out of range particles*/
+                    bool overflowEnabled;
+
+                    constexpr LogAxisKernel(T_AttrFunctor attrFunc, bool enableOverflowBins)
+                        : getAttributeValue{attrFunc}
+                        , overflowEnabled{enableOverflowBins}
+                    {
+                    }
+
+                    /**
+                     * @param n_bins User requested n_bins
+                     */
+                    void initAxisSplit(T logMinR, T logMaxR, uint32_t n_bins, ScalingType scalingR)
+                    {
+                        logMin = logMinR;
+                        logMax = logMaxR;
+                        nBins = n_bins;
+                        scaling = scalingR;
+                    }
+
+
+                    template<typename T_Worker, typename T_Particle>
+                    ALPAKA_FN_ACC uint32_t getBinIdx(
+                        const DomainInfo& domainInfo,
+                        const T_Worker& worker,
+                        const T_Particle& particle,
+                        bool& validIdx) const
+                    {
+                        auto val = getAttributeValue(domainInfo, worker, particle);
+
+                        static_assert(
+                            std::is_same<decltype(val), decltype(logMin)>::value,
+                            "The return type of the axisAttributeFunctor should be the same as the type of Axis "
+                            "min/max ");
+
+                        uint32_t binIdx = 0;
+                        bool enableBinning = overflowEnabled; // @todo check if disableBinning is better
+
+                        if(0. < val)
+                        {
+                            auto logVal = math::log2(val);
```

Comment:
for values < 1 `logVal` will be negative if I remember correctly you said that you must handle it in your implementation?

Reply:
@BrianMarre You were handling values from 0-1 as a special case, but I am not sure about the reason for this. I think we discussed that you deal with 0-1 separately to avoid a case where most log bins are in the 0-1 value range . From my point of view, we should do what the user asks, and if they want log bins with a minimum value which is negative we should let them. But I am not sure what would be expected/wanted behavior. Also pinging @steindev @PrometheusPi for input

Reply:
So this special handling will be moved to a different log-lin-log axis which has a log scales on either side of zero except for a user definable linear region around zero